ReSharper.FSharp/src/FSharp.Psi.Features/src/LanguageService/FSharpDeclaredElementIconProvider.fs

```diff
@@ -24,12 +24,15 @@ type FSharpDeclaredElementIconProvider() =
     let mutableInternalField = compose mutableField PsiSymbolsThemedIcons.ModifiersInternal.Id
 
     interface IDeclaredElementIconProvider with
-        member x.GetImageId(declaredElement, _, [<Out>] canApplyExtensions) =
+        member x.GetImageId(declaredElement, lang, [<Out>] canApplyExtensions) =
             canApplyExtensions <- true
 
             match declaredElement with
             | :? IFSharpModule -> FSharpIcons.FSharpModule.Id
 
+            | :? IFSharpGeneratedFromUnionCase as e ->
```

Comment:
This seems to affect more types than those representing union cases. Please limit it to union case representations.